src/lib.rs

```diff
@@ -267,77 +302,184 @@ fn interface_and_mtu_windows(socket: &UdpSocket) -> Result<(String, usize), Erro
-        #[cfg(target_os = "macos")]
-        check_mtu("localhost:443", false, 16384);
-        #[cfg(target_os = "linux")]
-        check_mtu("localhost:443", false, 65_536);
-        #[cfg(target_os = "windows")]
-        check_mtu("localhost:443", false, 4_294_967_295);
+    fn loopback_v4_loopback_v6() {
+        assert!(interface_and_mtu(&(local_v4(), local_v6())).is_err());
+    }
+
+    #[test]
+    fn loopback_v6_loopback_v4() {
+        assert!(interface_and_mtu(&(local_v6(), local_v4())).is_err());
+    }
+
+    #[test]
+    fn loopback_v6_loopback_v6() {
+        assert!(interface_and_mtu(&(local_v6(), local_v6())).unwrap() == LOOPBACK);
+    }
+    #[test]
+    fn none_loopback_v4() {
+        assert!(interface_and_mtu(&(None, local_v4())).unwrap() == LOOPBACK);
+    }
+
+    #[test]
+    fn none_loopback_v6() {
+        assert!(interface_and_mtu(&(None, local_v6())).unwrap() == LOOPBACK);
+    }
+
+    #[test]
+    fn loopback_v4_none() {
+        assert!(interface_and_mtu(&(local_v4(), None)).unwrap() == LOOPBACK);
+    }
+
+    #[test]
+    fn loopback_v6_none() {
+        assert!(interface_and_mtu(&(local_v6(), None)).unwrap() == LOOPBACK);
+    }
+
+    #[test]
+    fn inet_v4_inet_v4() {
+        assert!(interface_and_mtu(&(inet_v4(), inet_v4())).is_err());
+    }
+
+    #[test]
+    fn inet_v4_inet_v6() {
+        assert!(interface_and_mtu(&(inet_v4(), inet_v6())).is_err());
+    }
+
+    #[test]
+    fn inet_v6_inet_v4() {
+        assert!(interface_and_mtu(&(inet_v6(), inet_v4())).is_err());
+    }
+
+    #[test]
+    fn inet_v6_inet_v6() {
+        assert!(interface_and_mtu(&(inet_v6(), inet_v6())).is_err());
+    }
+    #[test]
+    fn none_inet_v4() {
+        assert!(interface_and_mtu(&(None, inet_v4())).unwrap() == INET);
+    }
+
+    #[test]
+    fn none_inet_v6() {
+        if env::var("CI").is_ok() {
+            // The GitHub CI environment does not have IPv6 connectivity.
+            return;
+        }
+        assert!(interface_and_mtu(&(None, inet_v6())).unwrap() == INET);
     }
 
     #[test]
-    fn default_interface_mtu_v4() {
-        check_mtu("ietf.org:443", true, 1500);
+    fn inet_v4_none() {
+        assert!(interface_and_mtu(&(inet_v4(), None)).is_err());
     }
 
     #[test]
-    fn default_interface_mtu_v6() {
-        check_mtu("ietf.org:443", false, 1500);
+    fn inet_v6_none() {
+        assert!(interface_and_mtu(&(inet_v6(), None)).is_err());
     }
 
     #[test]
     #[allow(deprecated)] // Purpose of the test is to cover deprecated functions.
     fn deprecated_functions() {
-        let addr = "localhost:443".to_socket_addrs().unwrap().next().unwrap();
-        assert!(super::interface_mtu(&addr).is_ok());
-        assert!(super::get_interface_mtu(&addr).is_ok());
+        assert!(super::interface_mtu(&local_v4()).is_ok());
+        assert!(super::get_interface_mtu(&local_v4()).is_ok());
+        assert!(interface_and_mtu(&local_v4()).is_ok());
     }
```

Comment:
I know that people love API stability, but this crate isn't that old.  Can we not just jettison the old API?

Reply:
WFM. @mxinden? @djc?

Reply:
Makes sense to me!

Reply:
Yes, fine by me!

Reply:
Thanks, I'll rip it out.